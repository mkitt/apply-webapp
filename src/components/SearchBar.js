// @flow
import React from 'react'
import glamorous from 'glamorous'
import AppContainment from './elements/AppContainment'
import { MagGlassIcon } from './elements/Icons'


type Props = {
  action: string,
  forId: string,
  handleChange: Function,
  handleSubmit: Function,
  value: string,
}

export default({ action, forId, handleChange, handleSubmit, value }: Props) => (
  <AppContainment marginBottom={40} >
    <Form
      acceptCharset="UTF-8"
      action={action}
      method="get"
      onSubmit={handleSubmit}
    >
      <Label htmlFor={forId}>
        <Icon><MagGlassIcon /></Icon>
        <Input
          aria-label="Search for applicants"
          autoCapitalize="off"
          autoComplete="off"
          autoCorrect="off"
          id={forId}
          onChange={handleChange}
          placeholder="Search Applicants"
          type="search"
          value={value}
        />
      </Label>
    </Form>
  </AppContainment>
)

// -------------------------------------

const Form = glamorous.form({
  position: 'relative',
  backgroundColor: '#fff',
})

const Label = glamorous.label({
  position: 'relative',
  display: 'flex',
  alignItems: 'center',
  color: '#808080',
})

const Input = glamorous.input({
  width: '100%',
  margin: 0,
  padding: 8,
  paddingLeft: 40,
  fontFamily: 'inherit',
  fontSize: 34,
  fontWeight: 200,
  color: 'inherit',
  backgroundColor: 'transparent',
  border: 0,
  borderBottom: '1px solid',
  outline: 'none',
  WebkitAppearance: 'textfield',
})

const Icon = glamorous.span({
  position: 'absolute',
  top: 12,
  left: 0,
})
