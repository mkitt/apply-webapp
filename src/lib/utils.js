// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
