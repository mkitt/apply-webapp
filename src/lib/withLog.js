import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
