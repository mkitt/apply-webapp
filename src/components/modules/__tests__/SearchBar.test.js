import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from '../../../lib/utils'
import SearchBar from '../SearchBar'

describe('<SearchBar />', () => {
  it('shows the correct snapshot tree for the <SearchBar> module (no value)', () => {
    const component = (
      <SearchBar
        action="/searchbar-test"
        forId="forId"
        handleChange={noop}
        handleSubmit={noop}
        value=""
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <SearchBar> module (with value)', () => {
    const component = (
      <SearchBar
        action="/searchbar-test-2"
        forId="forId2"
        handleChange={noop}
        handleSubmit={noop}
        value="Foo"
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
