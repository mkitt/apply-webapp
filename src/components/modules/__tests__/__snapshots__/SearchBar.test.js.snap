// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> shows the correct snapshot tree for the <SearchBar> module (no value) 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  max-width: 1392px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 8px;
  padding-bottom: 16px;
  padding-left: 8px;
  margin-bottom: 40px;
}

.css-1,
[data-css-1] {
  position: relative;
  background-color: #fff;
}

.css-2,
[data-css-2] {
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-top: 24px;
  color: #808080;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-3,
[data-css-3] {
  position: absolute;
  top: 12px;
  left: 0;
}

.css-4,
[data-css-4] {
  width: 100%;
  margin: 0;
  padding: 8px;
  padding-left: 40px;
  font-family: inherit;
  font-size: 34px;
  font-weight: 200;
  color: inherit;
  background-color: transparent;
  border: 0;
  border-bottom: 1px solid;
  outline: none;
  -webkit-appearance: textfield;
}

<div
  className="css-0"
>
  <form
    acceptCharset="UTF-8"
    action="/searchbar-test"
    className="css-1"
    method="get"
    onSubmit={[Function]}
  >
    <label
      className="css-2"
      htmlFor="forId"
    >
      <span
        className="css-3"
      >
        <svg
          className="css-nil"
          height={32}
          viewBox="0 0 28 28"
          width={32}
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
          >
            <g
              transform="translate(-40.000000, -119.000000)"
            >
              <g
                transform="translate(19.000000, 104.000000)"
              >
                <g
                  transform="translate(23.000000, 17.000000)"
                >
                  <path
                    d="M17.7401739,8.86956522 C17.7401739,3.97147826 13.7681739,0 8.86956522,0 C3.972,0 0,3.97147826 0,8.86956522 C0,13.7676522 3.972,17.7391304 8.86956522,17.7391304 C13.7681739,17.7391304 17.7401739,13.7676522 17.7401739,8.86956522 Z"
                  />
                  <path
                    d="M15.2007652,15.1997217 L23.9998957,23.9998957"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </span>
      <input
        aria-label="Search for applicants"
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="css-4"
        id="forId"
        onChange={[Function]}
        placeholder="Search Applicants"
        type="search"
        value=""
      />
    </label>
  </form>
</div>
`;

exports[`<SearchBar /> shows the correct snapshot tree for the <SearchBar> module (with value) 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  max-width: 1392px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 8px;
  padding-bottom: 16px;
  padding-left: 8px;
  margin-bottom: 40px;
}

.css-1,
[data-css-1] {
  position: relative;
  background-color: #fff;
}

.css-2,
[data-css-2] {
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-top: 24px;
  color: #808080;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-3,
[data-css-3] {
  position: absolute;
  top: 12px;
  left: 0;
}

.css-4,
[data-css-4] {
  width: 100%;
  margin: 0;
  padding: 8px;
  padding-left: 40px;
  font-family: inherit;
  font-size: 34px;
  font-weight: 200;
  color: inherit;
  background-color: transparent;
  border: 0;
  border-bottom: 1px solid;
  outline: none;
  -webkit-appearance: textfield;
}

<div
  className="css-0"
>
  <form
    acceptCharset="UTF-8"
    action="/searchbar-test-2"
    className="css-1"
    method="get"
    onSubmit={[Function]}
  >
    <label
      className="css-2"
      htmlFor="forId2"
    >
      <span
        className="css-3"
      >
        <svg
          className="css-nil"
          height={32}
          viewBox="0 0 28 28"
          width={32}
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
          >
            <g
              transform="translate(-40.000000, -119.000000)"
            >
              <g
                transform="translate(19.000000, 104.000000)"
              >
                <g
                  transform="translate(23.000000, 17.000000)"
                >
                  <path
                    d="M17.7401739,8.86956522 C17.7401739,3.97147826 13.7681739,0 8.86956522,0 C3.972,0 0,3.97147826 0,8.86956522 C0,13.7676522 3.972,17.7391304 8.86956522,17.7391304 C13.7681739,17.7391304 17.7401739,13.7676522 17.7401739,8.86956522 Z"
                  />
                  <path
                    d="M15.2007652,15.1997217 L23.9998957,23.9998957"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </span>
      <input
        aria-label="Search for applicants"
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="css-4"
        id="forId2"
        onChange={[Function]}
        placeholder="Search Applicants"
        type="search"
        value="Foo"
      />
    </label>
  </form>
</div>
`;
